enhancements/hybrid-helm-operator.md

```diff
@@ -0,0 +1,223 @@
+   utility library, to minimize the amount of code the user will need to maintain. This
+   library could either be the existing operator-lib package, or a new, helm-operator specific
+   package.
+
+### User Stories
+
+#### Story 1
+
+As a Helm-based operator author, I want to add additional programmatic backup and restore
+functionality to my operator, without starting over with a completely different type of operator.
+
+#### Story 2
+
+As a Helm-based operator author, I want to change core behavior of the reconciliation loop
+that is not currently exposed in the watches.yaml
+
+#### Story 3
+
+As a prospective author of a Helm-based operator, I want to know that if I invest time and
+effort creating a Helm-based operator, I won't need to start over in order to create a higher
+level operator.
+
+#### Story 4
+
+As a Golang-based operator author, I would like to leverage existing helm charts for deploying
+pieces of my application, without sacrificing the flexibility of Go.
+
+
+### Implementation Details/Notes/Constraints
+
+#### Scaffolding plugin for the SDK
+
+The basic idea behind the hybrid Helm/Go operator plugin is that we will scaffold out a Go
+operator project structure, with all of the code bits from the helm-operator project filled in,
+such that building the project without adding any APIs or modifying any code will result in the
+same container as what is shipped in the helm-operator base image.
+
+Because the driving need behind this functionality is the ability to customize any piece of
+the Helm-based operator, we will scaffold out every bit of code that is not accessible via
+a library, and err on the side of scaffolding more than we need to.
+
+For the first iteration of the hybrid Helm/Go operator plugin, we will likely just scaffold
+out nearly everything that is currently in the [helm-operator repository][helm-operator-repo], but
+with a `main.go` that is more in line with a typical Go-based operator, including the kubebuilder
+scaffolding markers that make it easy to add pure Go-based apis. This will provide maximal
+flexibility for early adopters of the pre-stable plugin. This also matches the functionality that
+current users of the [helm-operator repository][helm-operator-repo] have, as the way they have
+customized the behavior of the `helm-operator` binary is by forking the repository entirely and
+making the modifications in their fork.
+
+Examples of code that will likely have to remain scaffolded in the long-term are:
+1. `pkg/watches/watches.go`: Based on current usage of the [helm-operator repository][helm-operator-repo],
+    being able to modify the internal structure of the `watches.yaml` configuration file by adding options
+    or changing their handling is likely to be a long-term need.
+1. `internal/cmd/run/cmd.go`: Similarly to the `watches.go` file, users will need to have the
+   ability to freely modify the arguments the `helm-operator` binary takes.
```

Comment:
part of `pkg/reconciler/reconciler.go` too maybe? Because being able to add pre and post hooks to the helm reconciler is a useful feature of hybrid operator.